### PR TITLE
Adds a message that allows loading environments via a topic

### PR DESCRIPTION
**Pull request by arjo129**
# 🦟 Bug fix

Fixes #<NUMBER>

## Summary
Required by https://github.com/gazebosim/gz-sim/pull/1842

Signed-off-by: Arjo Chakravarty <arjo@openrobotics.org>
## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/gz/msgs/Utility.hh [hunk 1]
```diff
@@ -212,6 +220,14 @@ namespace gz
     msgs::SphericalCoordinates Convert(
         const math::SphericalCoordinates &_coord);
 
+    /// \brief Convert a msgs::SphericalCoordinatesType to an
+    /// gz::math::SphericalCoordinates::CoordinateTpye
+    /// \param[in] _coord The spherical coordinates to convert
+    /// \return A gz::math::SphericalCoordinatesType object
+    GZ_MSGS_VISIBLE
+    msgs::SphericalCoordinatesType ConvertCoord(
+      const math::SphericalCoordinates::CoordinateType &_sc);
```
Comment (ahcorde) [inline include/gz/msgs/Utility.hh:229]:
```suggestion
    /// \param[in] _coord The spherical coordinates to convert
    /// \return A gz::math::SphericalCoordinatesType object
    GZ_MSGS_VISIBLE
    msgs::SphericalCoordinatesType ConvertCoord(
      const math::SphericalCoordinates::CoordinateType &_coord);
```

File: proto/gz/msgs/data_load_options.proto [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+package gz.msgs;
+option java_package = "com.gz.msgs";
+option java_outer_classname = "DataLoadPathOptions";
+
+import "gz/msgs/spherical_coordinates.proto";
+
+
+
+/// \brief Used for specifying how to load environmental data
+message DataLoadPathOptions
```
Comment (ahcorde) [inline proto/gz/msgs/data_load_options.proto:28]:
```suggestion
import "gz/msgs/spherical_coordinates.proto";

/// \brief Used for specifying how to load environmental data
message DataLoadPathOptions
```

**Review by ahcorde — DISMISSED** [review]

File: include/gz/msgs/Utility.hh [hunk 3]
```diff
@@ -212,6 +220,14 @@ namespace gz
     msgs::SphericalCoordinates Convert(
         const math::SphericalCoordinates &_coord);
 
+    /// \brief Convert a msgs::SphericalCoordinatesType to an
+    /// gz::math::SphericalCoordinates::CoordinateTpye
+    /// \param[in] _coord The spherical coordinates to convert
```
Comment (caguero) [inline include/gz/msgs/Utility.hh:225]: The spherical coordinate type ...

File: include/gz/msgs/Utility.hh [hunk 4]
```diff
@@ -212,6 +220,14 @@ namespace gz
     msgs::SphericalCoordinates Convert(
         const math::SphericalCoordinates &_coord);
 
+    /// \brief Convert a msgs::SphericalCoordinatesType to an
```
Comment (caguero) [inline include/gz/msgs/Utility.hh:223]: The other way, from `math::` to `msgs::` .

File: proto/gz/msgs/data_load_options.proto [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+package gz.msgs;
+option java_package = "com.gz.msgs";
+option java_outer_classname = "DataLoadPathOptions";
+
+import "gz/msgs/spherical_coordinates.proto";
+
+/// \brief Used for specifying how to load environmental data
+message DataLoadPathOptions
+{
+    /// \brief
```
Comment (caguero) [inline proto/gz/msgs/data_load_options.proto:28]: Add doc.

File: src/Utility.cc [hunk 6]
```diff
@@ -183,6 +183,25 @@ namespace gz
...
+        case msgs::SphericalCoordinatesType::ECEF:
+          return math::SphericalCoordinates::CoordinateType::ECEF;
+        case msgs::SphericalCoordinatesType::GLOBAL:
+          return math::SphericalCoordinates::CoordinateType::GLOBAL;
+        case msgs::SphericalCoordinatesType::SPHERICAL:
+          return math::SphericalCoordinates::CoordinateType::SPHERICAL;
+        case msgs::SphericalCoordinatesType::LOCAL:
+          return math::SphericalCoordinates::CoordinateType::LOCAL;
+        case msgs::SphericalCoordinatesType::LOCAL2:
+          return math::SphericalCoordinates::CoordinateType::LOCAL2;
```
Comment (caguero) [inline src/Utility.cc:201]: This block returns a warning on Windows as we don't return anything if none of the previous conditions are met.

File: src/Utility.cc [hunk 7]
```diff
@@ -349,6 +368,25 @@ namespace gz
...
+        case math::SphericalCoordinates::CoordinateType::ECEF:
+          return msgs::SphericalCoordinatesType::ECEF;
+        case math::SphericalCoordinates::CoordinateType::GLOBAL:
+          return msgs::SphericalCoordinatesType::GLOBAL;
+        case math::SphericalCoordinates::CoordinateType::SPHERICAL:
+          return msgs::SphericalCoordinatesType::SPHERICAL;
+        case math::SphericalCoordinates::CoordinateType::LOCAL:
+          return  msgs::SphericalCoordinatesType::LOCAL;
+        case math::SphericalCoordinates::CoordinateType::LOCAL2:
+          return msgs::SphericalCoordinatesType::LOCAL2;
```
Comment (caguero) [inline src/Utility.cc:386]: This block returns a warning on Windows as we don't return anything if none of the previous conditions are met.

File: proto/gz/msgs/data_load_options.proto [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        DEGREES = 1;
+    }
+
+    /// \brief File path to load
+    string path = 1;
+
+    /// \brief Name of time axis
+    string time = 2;
+
+     /// \brief Is the data static in time
```
Comment (caguero) [inline proto/gz/msgs/data_load_options.proto:41]: Nit: extra space here.

**Review by caguero — APPROVED** [review]
Tiny comment, otherwise looks good to me.

**Comment by arjo129** [comment]
I suggest holding off on merging this till gazebosim/gz-sim#1842 is complete.

File: proto/gz/msgs/data_load_options.proto [hunk 9]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
```
Comment (ahcorde) [inline proto/gz/msgs/data_load_options.proto:2]:
```suggestion
 * Copyright (C) 2023 Open Source Robotics Foundation
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Review by mjcarroll — CHANGES_REQUESTED** [review]
There are some open comments from Carlos here, if you could address them.

**Review by ahcorde — APPROVED** [review]

**Comment by arjo129** [comment]
@mjcarroll I've addressed the issues. It's waiting on a :heavy_check_mark: from you.

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**